### DISCUSSION TITLE
MWP on MacOS Sonoma

### DISCUSSION THREAD

**Discussion by silicongarage** [stronnag/mwptools#180] [Q&A]
I'm trying to build on MacOS and have gotten farther than #135 .   I've hit a wall with libepoxy.  I get the following error:

`../src/mwp/meson.build:76:35: ERROR: C shared or static library 'epoxy' not found`

Homebrew only installs a dynamic lib version of libepoxy, so I downloaded the source (v1.5.10) and managed to build and install  static libepoxy.a using the --default-library=static switch.  However I still get the above mwp meson message about the static library not being found.   At this point I'm not sure how to proceed.

**Comment by stronnag** [comment]
I had wondered whether it would be possible to build the GTK4 version on MacOS. Congratulations on getting this far. 

I would be interested to see the whole  Meson output up to failure. That will give me some clue as how to proceed. 

Meson will use `pkg-config` to determine if a dependency exists. What is the output of:
```
$ pkg-config  --print-provides epoxy
``` 
If there is none, or an error it is possible that the HomeBrew recipe either installed it elsewhere  or not at all. 
As libepoxy  is not entirely essential, I could modify mwp to only optionally require it. 

But please provide the above information (meson output up to failure and pkg-config for epoxy) before I go that route.

**Comment by stronnag** [comment]
On second thoughts, it's maybe more complex that than, as iirc, Homebrew uses `/opt/<something>` vice `/usr/local/lib`. So even if I remove `libepoxy` it will fail other pure C libraries.

**Comment by stronnag** [comment]
Indeed, I need to rework the way mwp evinces pure C libraries.  I'll let you know then it's done.

**Comment by stronnag** [comment]
There is now a new development branch `meson_libs` which changes now meson handles C libraries.
```
git pull
git checkout meson-libs
```

Then I suggest you reconfigure with a new build directory, e.g.:
```
meson  setup _xbuild --prefix=~/.local --strip 
ninja -C _xbuild install
```
If it fails to find required dependencies, you can specify extra library paths (e.g. on my FreeBSD box);
```
rm -rf _xbuild
meson  setup _xbuild --prefix=~/.local --strip -Dextra-lib-paths=/usr/local/lib
# etc ...
```
The `-Dextra-lib-paths=` argument takes a comma separated list:
e.g
```
-Dextra-lib-paths=/usr/local/lib,/opt/homebrew/libs`
```

**Reply by silicongarage** [reply]
The output of `$ pkg-config  --print-provides epoxy` returns `epoxy = 1.5.10`
I did a pull of meson_libs but now get a dependency for blueprint-compiler

`Program blueprint-compiler found: NO`
`data/meson.build:38:12: ERROR: Program 'blueprint-compiler' not found or not executable`

My searching shows Blueprint not available on Homebrew and couldn't find any info via Google on MacOS version

**Comment by stronnag** [comment]
It's a python program, so it probably runs everywhere. Certainly does on FreeBSD.

**Comment by stronnag** [comment]
https://gitlab.gnome.org/jwestman/blueprint-compiler

**Comment by stronnag** [comment]
In another [project](https://github.com/stronnag/fl2xui/) (which builds on macos), I added `blueprint-compiler` as a sub-project. 
In the process of doing that ....

**Comment by stronnag** [comment]
If you pull meson-libs again, it will install its own `blueprint-compiler` as a sub-project.
You may then find you're missing some python dependencies ...

**Comment by stronnag** [comment]
Apropos the (now missing) message re arm64 symbols, potential workaround just pushed to `meson-libs`.

**Comment by stronnag** [comment]
And now a really ugly workaround for the compiler stuff ...

**Reply by silicongarage** [reply]
here's the latest output:
[ninja.log](https://github.com/user-attachments/files/17820499/ninja.log)

**Comment by stronnag** [comment]
That looks like it builds ... 
did you install it ?
does it run ?

**Comment by silicongarage** [comment]
woo hoo!
<img width="2304" alt="Screenshot 2024-11-19 at 1 54 00 PM" src="https://github.com/user-attachments/assets/a13a120a-47a9-4b6b-b4bb-013f86d347f5">

**Comment by stronnag** [comment]
Well ... I'm shocked. Some problems with the panel scaling (which is meant to fit a mono-spaced "00:00:00.0N 000:00:00.0W", but amazing all the same.

**Reply by silicongarage** [reply]
Thanks for your persistence!   I apologize about removing the "arm64" message.  Was going to update with more info but you were too quick and saw the original.    Other than the epoxy & blueprint issues, most of the build on my end was easy, just picking up missing deps with brew.

**Comment by stronnag** [comment]
So in summary and so I don't forget,  I need to:
* pick out the C library dependencies (inc. removal of libepoxy).
* include the sub-project blueprint-compiler
* fix the extraction of the compiler metadata
* fix the 'weak' symbol in the speech wrapper
* add a setting for minimum panel size.

then anything else that doesn't work on MacOS ...

**Comment by silicongarage** [comment]
New question:
Is it possible to add support for BLE (like iNav) with  fc-get & fc-set.    I use Speedybee F405s (wing & mini) BLE for connectivity.

**Reply by stronnag** [reply]
> New question: Is it possible to add support for BLE (like iNav) with fc-get & fc-set. I use Speedybee F405s (wing & mini) BLE for connectivity.

unfortunatly, BLE is completely OS specific, so no. However I expect there are macos BLE tools (that bridge ble into either a pseudo-terminal or IP connection.

**Comment by stronnag** [comment]
At your convenience, could you please, from the `mwptools` project directory, run the following:
```
src/mwp/getinfo.sh /tmp/vers.txt
```
and let me have the resulting `/tmp/vers.txt` (this is build info that the MacOS compiler did not like).

Thanks.

**Comment by stronnag** [comment]
and if I could also trouble you for:
```
clang  -dM -E -  </dev/null > /tmp/clangdefs.txt
```
i.e. the resulting `/tmp/clangdefs.txt`

**Reply by silicongarage** [reply]
[vers.txt](https://github.com/user-attachments/files/17821520/vers.txt)
[clangdefs.txt](https://github.com/user-attachments/files/17821521/clangdefs.txt)

**Comment by stronnag** [comment]
Thank you.

In theory, the `development` branch has been updated with more correct versions of everything above.
It also adds a setting for the sidebar width.
```
# zero means don't scale
gsettings get org.stronnag.mwp sidebar-scale-factor
0.0
# make 20% wider
gsettings set org.stronnag.mwp sidebar-scale-factor 1.2
```

**Reply by silicongarage** [reply]
I created a new dir and did a clone from your repo.
Getting the epoxy error again.
`src/mwp/meson.build:76:35: ERROR: C shared or static library 'epoxy' not found`

**Comment by stronnag** [comment]
Damn, must have missed that one.

**Comment by stronnag** [comment]
Or your haven't checked out `development` ....

**Comment by silicongarage** [comment]
ok, I was on master

**Comment by stronnag** [comment]
Easily done ...

**Reply by silicongarage** [reply]
meson builds, but getting an ninja error - maybe something I broke on my end

[ninja.txt](https://github.com/user-attachments/files/17822056/ninja.txt)

**Reply by silicongarage** [reply]
I had built blueprintcompiler from the source link, looks like it installed it into the homebrew site path

**Comment by stronnag** [comment]
You're using a homebrew installed blueprint compiler rather than the compiler bundled with mwp in this log.

It is possible that the homebrew installed blueprint compiler looks for resources in the homebrew directories rather than system ones. 

This is your problem --- you're right --- you broke it!!

**Reply by silicongarage** [reply]
I'm looking how to remove it now

**Comment by stronnag** [comment]
Once you've removed it, you'll need to run `meson` again.

**Reply by silicongarage** [reply]
okay, now its using the subproject, but still ninja error
[ninja.txt](https://github.com/user-attachments/files/17822269/ninja.txt)

**Comment by stronnag** [comment]
The interesting part is (same in both cases):
```
    import gi  # type: ignore
    ^^^^^^^^^
ModuleNotFoundError: No module named 'gi'
```
which is a python module often known as `PyGObject`. 

At some stage you must have installed it (maybe as a side-effect of installing some other dependency) and you've subsequently managed to "uninstall" it.  No idea how.

**Reply by silicongarage** [reply]
I had a temp python env, must have been in there.  Anyway I did a brew install of PyGOject.
 Now when I run meson it complains about libsoup-3.0

**Comment by stronnag** [comment]
No idea, these dependencies were obviously present earlier. mwp doesn't remove them.

**Reply by silicongarage** [reply]
seems I installed libsoup with brew,  need to use conda

**Comment by silicongarage** [comment]
At this point, I'm surprised I got to work the first time! 
GTK4 problem.    Looking for version >= 4.14 but found 4.12.1   Thats the latest via conda.  When I install with brew I get 4.16.5 but meson doesn't find it.

**Comment by silicongarage** [comment]
<img width="2304" alt="image" src="https://github.com/user-attachments/assets/668f0cf5-0f94-441a-a8d0-0bc3b7341e29">

**Reply by silicongarage** [reply]
`% gsettings get org.stronnag.mwp sidebar-scale-factor`
`1.2`

**Comment by stronnag** [comment]
OK, so it seems that Gtk4/MacOS/Quartz does not recognise the sizing of the side bar. There are two possibilities:
* The so called  `LengthUnit` is not recognised or
* It's just broken on MacOS.

We can probably address both of these issues, starting with the easiest (the first).

* I've setup a `macos_testing` branch. It is not necessary to re-clone the repo; just doing:
```
git pull
git checkout macos_testing
```
will suffice.

Note also that mwp generates a log in your home directory `~/mwp_stderr_YYYY-MM-DD.txt` (e.g. `mwp_stderr_2024-11-20.txt`). This will contain debug information that I'd like to see. So at the end , please attach the log file.

The tests are controlled by two environment variables; I'm not familiar with setting env vars in the MacOS default `zsh`, but I think prefixing commands with `env` will work.

## Prerequisite

* `macos_testing` pulled, built, installed
* Reset 	`sidebar-scale-factor`

```
gsettings set org.stronnag.mwp sidebar-scale-factor 0
```

## First Test

```
env MWP_LENGTH_UNIT=0 mwp
```
Let me know if this fixes the sidebar (such that the location bit  looks something like the below (the position is all visible).

![image](https://github.com/user-attachments/assets/4ab32be2-38aa-4d87-8cb3-ba7fe709bbb9)

## Second Test

```
env MWP_LENGTH_UNIT=1 mwp
```
Again, only interested if the sidebar is correctly sized (or not).

## Third Test

This test places the sidebar in a "paned" widget. It will allow the side bar to be sized by moving the divider between it and the map, though I rather hope that it might be correctly sized anyway.

```
env MWP_USE_PANED=1 mwp
```

An image and the log file would be appreciated here; I'm rather hopeful this test will work; however there may be some undesired side effects.

And if you decide it's all too much, that's fine too. However, having got this far I'd really like to have something usable on MacOS.

**Reply by silicongarage** [reply]
I'm more than happy to help

**Comment by silicongarage** [comment]
First Test
% export MWP_LENGTH_UNIT=0
<img width="289" alt="image" src="https://github.com/user-attachments/assets/10a5f7f8-d8df-4731-9897-7cc3d8f65f16">

**Reply by stronnag** [reply]
OK, expected

**Comment by silicongarage** [comment]
Second Test
% export  MWP_LENGTH_UNIT=1
<img width="286" alt="image" src="https://github.com/user-attachments/assets/e11f8e49-666f-4443-bac2-ba3ca1d28f9e">

**Reply by stronnag** [reply]
OK, I had a small hope, but not surprised. Everything rides on PANED ...

**Comment by silicongarage** [comment]
Third Test
% export  MWP_USE_PANED=1
<img width="424" alt="image" src="https://github.com/user-attachments/assets/4e5ef3ec-8275-40ea-9e46-929e6d947f9c">
sized by dragging the divider

**Comment by silicongarage** [comment]
[mwp_stderr_2024-11-20.txt](https://github.com/user-attachments/files/17835315/mwp_stderr_2024-11-20.txt)

**Comment by stronnag** [comment]
We have a winner. You can adjust the pane size (and the internal panes) to remove excess space.
I can then (TBD) save the pane width and reload it on next invocation. 
Cool. It looks like we have a viable MacOS build.

So the next step will be to set the panel implementation depending on whether the OS is MacOS or other.

**Comment by silicongarage** [comment]
ha, didn't know you could adjust inner panes!
<img width="289" alt="image" src="https://github.com/user-attachments/assets/24b12730-c466-4f1d-9a2f-5023beca1efb">

**Reply by stronnag** [reply]
Strangely, there is a (comprehensive) [user guide](https://stronnag.github.io/mwptools) and specifically for the [panel]( https://stronnag.github.io/mwptools/mwp-Gtk4-migration-guide/#side-panel).

**Comment by stronnag** [comment]
All the MacOS related changes are now merged into the `development` branch, including:
* Panel is automatically set to "Paned" on MacOS
* The pane size is saved / restored.

Looking forward to:
* (Seriously) issues for other things that may be broken on MacOS
* A PR (please, please) documenting how to build in MacOS. It could be as simple as "install this list of Hombrew packages, then follow the generic `meson` / `ninja` procedures. A PR would be cool, but even a simple text file I can merge into the docs would be  much appreciated.

**Reply by silicongarage** [reply]
My github experience is low, never have done a PR.   
Did so many pkg installs with mamba/conda and brew, need to figure out what a clean base is and then can document starting over.   One of the biggest issues I had  seems my env for homewbrew somehow kept get messed up.  Lots of the meson issues with libraries would be resolved by  redoing: 
`eval "$(/opt/homebrew/bin/brew shellenv)"` 

Given all that I will try to put something together.   As for other issues,  I don't have blackbox data, but use iNav->elrs->edgetx logs I've tried to replay.  The menu for `Replay OTX`  is grayed out.  Starting  with the csv file on the command line:
`mwp  flight-log.csv`
will prompt as preparing to replay, but then nothing ever happens.  Ha, probably just operator error.

**Reply by stronnag** [reply]
> will prompt as preparing to replay, but then nothing ever happens. Ha, probably just operator error.

Check the log, it will most likely tell you the dependency that is not installed.

**Comment by silicongarage** [comment]
latest development branch errors at runtime:
<img width="1220" alt="image" src="https://github.com/user-attachments/assets/fdc7d344-8410-4b26-b45d-c5a5f2193980">

**Comment by stronnag** [comment]
This typically happens if you install a version of mwp as a system (e.g. `/usr/local`) install, the do a later  install (after the settings schema has changed) as a local install (`~/.local`).

As the system wide settings database (by default) has priority, the new binary finds the old schema and is fatally unhappy.

To fix this, you need to find any obsolete versions of `org.stronnag.mwp.gschema.xml`, delete them and rebuild the associated settings database.

So, if the obsolete version was `/usr/local/share/glib-2.0/schemas/org.stronnag.mwp.gschema.xml`,
then you would:

```
### AS ROOT if it's a system directory ###
MPREFIX=/usr/local
sudo rm $MPREFIX/share/glib-2.0/schemas/org.stronnag.mwp.gschema.xml
sudo glib-compile-schemas $MPREFIX/share/glib-2.0/schemas
```

Adjust `MPREFIX` as needed, `sudo` as necessary.

Which reminds me that the "Uninstall" section of INSTALL script refers to some obsolete files; I need to fix that.

**Comment by silicongarage** [comment]
At least on my mac, seems  homebrew and conda/mamba don't play well together.   I resolved a lot of the meson build issues by making sure in my .zshrc that the homebrew setup is last insuring that it then shows up first in the search PATH before the conda dirs.

Still chasing the gsettings problem:
<img width="933" alt="image" src="https://github.com/user-attachments/assets/62249960-d728-42f5-8a54-194969ce05cd">

Looking at the last output from ninja install shows that glib-compile-schemas is being run:
<img width="1135" alt="image" src="https://github.com/user-attachments/assets/c0f3155e-ae95-4467-8d32-3d0a8ce6c5e7">

The schemas dir looks okay:
`% ls -ld /opt/local/share/glib-2.0/schemas/*`
`-rw-r--r--  1 buster  wheel   5656 Nov 22 10:04 /opt/local/share/glib-2.0/schemas/gschemas.compiled`
`-rw-r--r--  1 buster  wheel  24320 Nov 21 21:10 /opt/local/share/glib-2.0/schemas/org.stronnag.mwp.gschema.xml`

Syntax for gsettings on mac slightly different:
`Usage:`
 `gsettings --version`
  `gsettings [--schemadir SCHEMADIR] COMMAND [ARGS…]`

Running `% gsettings list-schemas`  doesn't show `org.stronnag.mwp` as an installed schema.
However `% gsettings --schemadir /opt/local/share/glib-2.0/schemas list-keys org.stronnag.mwp` works.

Running as `% sudo glib-compile-schemas /opt/local/share/glib-2.0/schemas`  doesn't make any difference.

**Comment by stronnag** [comment]
So you've somehow manged to install the setting schema in a place that the runtime doesn't know about (and for the record, `gsettings` is *exactly* the same on MacOS as it is anywhere else ...).

I have no visibility of what you've done,  so where is mwp installed (i.e. the ninja line that installed it).?
And what does `which mwp ` report?

**Comment by stronnag** [comment]
Also, what is the output of:
```
sudo find / -iname mwp
sudo find / -iname org.stronnag.mwp.gschema.xml
```
I'm interested to see if there are multiple examples outside of the build directory. If there are, which is the most recent?

**Comment by silicongarage** [comment]
I finally got it working.   

Finder searching for both `mwp` and `org.stronnag.mwp.gschema.xml` showed no strays.
For completeness:
`% which mwp`
`/opt/local/bin/mwp`

`% ls /opt/local/share/glib-2.0/schemas/*`
`/opt/local/share/glib-2.0/schemas/gschemas.compiled`
`/opt/local/share/glib-2.0/schemas/org.stronnag.mwp.gschema.xml`

From its behavior, it appears `mwp `is using the default schemas path and not the installed script path of `/opt/local/share/glib-2.0/schemas`

How `gsettings` path defaults on macos changes depending on how it is installed - different for homebrew and conda/mamba.

For homebrew the default database:
`/opt/homebrew/share/glib-2.0/schemas/gschemas.compiled`

and for mamba/conda the default database:
`/Users/buster/miniforge3/envs/default/share/glib-2.0/schemas/gschemas.compiled` where "default" is the name of the active env

With my current PATH structure in .zshrc:
`% which gsettings`
`/opt/homebrew/bin/gsettings`

I softlinked `/opt/local/share/glib-2.0/schemas/org.stronnag.mwp.gschema.xml ` into `/opt/homebrew/share/glib-2.0/schemas` and then ran `% glib-compile-schemas /opt/homebrew/share/glib-2.0/schemas`

Now `org.stronnag.mwp` shows up in the list when running `% gsettings list-schemas` and `mwp` starts up correctly

**Comment by silicongarage** [comment]
update: (mwptools 24.12.15)
Version builds and starts up.

steps:
(1) `% meson setup build --wipe --buildtype=release --strip --prefix /opt/local`
(2) `% ninja -C build install `
(3) `% glib-compile-schemas /opt/homebrew/share/glib-2.0/schemas`

There were some strange dylib warnings, don't know if that impacts anything:

> objc[39264]: Class ResultReceiver is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.43/lib/libgtk-3.0.dylib (0x1033f9e00) and /opt/homebrew/Cellar/gtk4/4.16.7/lib/libgtk-4.1.dylib (0x16fa2f270). One of the two will be used. Which one is undefined.
objc[39264]: Class GtkApplicationQuartzDelegate is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.43/lib/libgtk-3.0.dylib (0x1033f9e28) and /opt/homebrew/Cellar/gtk4/4.16.7/lib/libgtk-4.1.dylib (0x16fa2f298). One of the two will be used. Which one is undefined.
objc[39264]: Class GNSMenuItem is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.43/lib/libgtk-3.0.dylib (0x1033f9e78) and /opt/homebrew/Cellar/gtk4/4.16.7/lib/libgtk-4.1.dylib (0x16fa2f2e8). One of the two will be used. Which one is undefined.
objc[39264]: Class GNSMenu is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.43/lib/libgtk-3.0.dylib (0x1033f9ea0) and /opt/homebrew/Cellar/gtk4/4.16.7/lib/libgtk-4.1.dylib (0x16fa2f310). One of the two will be used. Which one is undefined.
objc[39264]: Class FilterComboBox is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.43/lib/libgtk-3.0.dylib (0x1033f9f18) and /opt/homebrew/Cellar/gtk4/4.16.7/lib/libgtk-4.1.dylib (0x16fa2f388). One of the two will be used. Which one is undefined.
objc[39264]: Class gdkCoreCursor is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.43/lib/libgdk-3.0.dylib (0x102745628) and /opt/homebrew/Cellar/gtk4/4.16.7/lib/libgtk-4.1.dylib (0x16fa2f3d8). One of the two will be used. Which one is undefined.

**Reply by stronnag** [reply]
At what stage do you see the dylib warnings?
You can see which libraries mwp is using using `otool`; it should not reference gtk+3.

If it runs then the warnings are probably bogus.

**Reply by silicongarage** [reply]
It only happened the first time running mwp after the build.
>(mwp) buster@Busters-MacBook-Pro mwptools % mwp
objc[39264]: Class ResultReceiver is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.43/lib/libgtk-3.0.dylib (0x1033f9e00) and /opt/homebrew/Cellar/gtk4/4.16.7/lib/libgtk-4.1.dylib (0x16fa2f270). One of the two will be used. Which one is undefined.
objc[39264]: Class GtkApplicationQuartzDelegate is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.43/lib/libgtk-3.0.dylib (0x1033f9e28) and /opt/homebrew/Cellar/gtk4/4.16.7/lib/libgtk-4.1.dylib (0x16fa2f298). One of the two will be used. Which one is undefined.

here's the output of otool:
>% otool -L /opt/local/bin/mwp
/opt/local/bin/mwp:
	/opt/local/lib/libmwpclib.3.dylib (compatibility version 3.0.0, current version 3.0.0)
	/opt/local/lib/libmwpvlib.3.dylib (compatibility version 3.0.0, current version 3.0.0)
	/opt/homebrew/opt/glib/lib/libglib-2.0.0.dylib (compatibility version 8201.0.0, current version 8201.2.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1351.0.0)
	/opt/homebrew/opt/gtk4/lib/libgtk-4.1.dylib (compatibility version 1601.0.0, current version 1601.7.0)
	/opt/homebrew/opt/pango/lib/libpango-1.0.0.dylib (compatibility version 5501.0.0, current version 5501.0.0)
	/opt/homebrew/opt/gdk-pixbuf/lib/libgdk_pixbuf-2.0.0.dylib (compatibility version 4201.0.0, current version 4201.12.0)
	/opt/homebrew/opt/cairo/lib/libcairo.2.dylib (compatibility version 2.0.0, current version 2.0.0)
	/opt/homebrew/opt/graphene/lib/libgraphene-1.0.0.dylib (compatibility version 1001.0.0, current version 1001.8.0)
	/opt/homebrew/opt/glib/lib/libgio-2.0.0.dylib (compatibility version 8201.0.0, current version 8201.2.0)
	/opt/homebrew/opt/glib/lib/libgobject-2.0.0.dylib (compatibility version 8201.0.0, current version 8201.2.0)
	/opt/homebrew/opt/libadwaita/lib/libadwaita-1.0.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/homebrew/opt/libsoup/lib/libsoup-3.0.0.dylib (compatibility version 8.0.0, current version 8.1.0)
	/usr/lib/libxml2.2.dylib (compatibility version 10.0.0, current version 10.9.0)
	/opt/homebrew/opt/libshumate/lib/libshumate-1.0.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/homebrew/opt/gstreamer/lib/libgstreamer-1.0.0.dylib (compatibility version 2411.0.0, current version 2411.0.0)
	/opt/homebrew/opt/json-glib/lib/libjson-glib-1.0.0.dylib (compatibility version 1001.0.0, current version 1001.0.0)
	/opt/homebrew/opt/libsecret/lib/libsecret-1.0.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/homebrew/opt/vte3/lib/libvte-2.91-gtk4.0.dylib (compatibility version 0.0.0, current version 0.0.0)
	/opt/homebrew/opt/protobuf-c/lib/libprotobuf-c.1.dylib (compatibility version 2.0.0, current version 2.0.0)

I've noticed that mwp is not remembering the pane position after I have moved them into position then quit and restarted

**Reply by stronnag** [reply]
> I've noticed that mwp is not remembering the pane position after I have moved them into position then quit and restarted

Vertically, horizontally or both ?

**Reply by silicongarage** [reply]
~~both~~ horizontal

**Reply by stronnag** [reply]
I have probably broken that when I changed mwp to save its size of exit.

Can you please:

```
gsettings set org.stronnag.mwp  p-pane-width 0
```
restart mwp and see if the size is saner.
I still think there is a problem in that the new panel width is not saved if you resize it.

**Reply by silicongarage** [reply]
horizontal still not remembered

**Reply by stronnag** [reply]
I expected that. OK, I will look at it.

**Comment by stronnag** [comment]
Just pushed an update to development that should save the pane location. It will also 'spam' the `mwp_stderr` log with lines like
```
21:02:16.991453 :DBG: leave paned ww=1499 ppos=1155 pw=344
```
when you drag the pane divider.
Hopefully this will fix the MacOS pane behaviour.

**Reply by silicongarage** [reply]
I have pulled the latest dev branch, and did a wipe/rebuild

>% git status
On branch development
Your branch is up to date with 'origin/development'.
nothing to commit, working tree clean

Still not saving the horizontal position
Here's the log file:
[mwp_stderr_2024-12-17.txt](https://github.com/user-attachments/files/18168012/mwp_stderr_2024-12-17.txt)

**Reply by silicongarage** [reply]
Maybe this is useful, I looked at the error in the log and tried to replicate on the CLI.
>% dbus-launch --autolaunch=453906ec9a7d0a84418bd2d26472e7d5 --binary-syntax --close-stderr
Autolaunch requested, but X11 support not compiled in.
Cannot continue.
% which dbus-launch
/opt/homebrew/bin/dbus-launch

**Reply by stronnag** [reply]
There is definitely a problem on MacOS with Gtk's mechanism. It's on my to do list.

**Comment by stronnag** [comment]
I finally made a new "hackitosh" VM (running Ventura). It confirms multiple issues here that will require special handling in the code:
* `gsettings` from the shell and `mwp` use different `gsettings` stores, so changing `mwp` setings via `gsettings` is futile. 
* MacOS /Gtk has a different idea of font size / string length in "pixels" compared to other platforms
* And some other issues.

**Comment by stronnag** [comment]
See #188
